addons/sale/models/analytic.py

```diff
@@ -0,0 +1,145 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import api, fields, models, _
+from odoo.exceptions import UserError
+
+
+class AccountAnalyticLine(models.Model):
+    _inherit = "account.analytic.line"
+
+    so_line = fields.Many2one('sale.order.line', string='Sales Order Line')
+
+    @api.model
+    def create(self, values):
+        result = super(AccountAnalyticLine, self).create(values)
+        result._sale_postprocess()
+        return result
+
+    @api.multi
+    def write(self, values):
+        # get current so lines for which update qty wil be required
+        sale_order_lines = self.env['sale.order.line']
+        if 'so_line' in values:
+            sale_order_lines = self.sudo().mapped('so_line')
+        result = super(AccountAnalyticLine, self).write(values)
+        self._sale_postprocess(values, force_so_lines=sale_order_lines)
+        return result
+
+    @api.multi
+    def unlink(self):
+        sale_order_lines = self.sudo().mapped('so_line')
+        res = super(AccountAnalyticLine, self).unlink()
+        self.env['account.analytic.line']._sale_postprocess(force_so_lines=sale_order_lines)
+        return res
+
+    @api.multi
+    def _sale_postprocess(self, values=None, force_so_lines=None):
```

Comment:
I would change a bit the flow to simplify the code here. What I propose
1/ _sale_postprocess called in create and write with given values, ok (do not forget to give values at create)
2/ in unlink: call directly sale_order_lines.sudo()._analytic_compute_delivered_quantity() as all code will always be avoided in the call chain
3/ remove force_so_lines argument in _sale_recompute_qty_delivered
4/ I think we can merge _sale_postprocess and _sale_recompute_qty_delivered to simplify the call chain